[SPARK-17524] [TESTS] Use specified spark.buffer.pageSize

a-roberts opened the pull request:
## What changes were proposed in this pull request?

This PR has the appendRowUntilExceedingPageSize test in RowBasedKeyValueBatchSuite use whatever spark.buffer.pageSize value a user has specified to prevent a test failure for anyone testing Apache Spark on a box with a reduced page size. The test is currently hardcoded to use the default page size which is 64 MB so this minor PR is a test improvement
## How was this patch tested?

Existing unit tests with 1 MB page size and with 64 MB (the default) page size

a-roberts committed:
Use specified spark.buffer.pageSize

Users can change the spark.buffer.pageSize value to run Spark unit tests on machines with less powerful hardware e.g. with two cores

To address this we can use whatever they've specified to avoid a failure where we see the number of rows for the appendRowUntilExceedingPageSize test is not as expected

SparkQA commented:
**[Test build #65319 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/65319/consoleFull)** for PR 15079 at commit [`c793efe`](https://github.com/apache/spark/commit/c793efe6e567ba3fc51ceaf9ff225ca1e5f35f10).

srowen commented on sql/catalyst/src/test/java/org/apache/spark/sql/catalyst/expressions/RowBasedKeyValueBatchSuite.java:
```diff
@@ -338,15 +338,18 @@ public void appendRowUntilExceedingCapacity() throws Exception {
   public void appendRowUntilExceedingPageSize() throws Exception {
+    // Use of this property prevents issues when running on a two core machine
+    // A developer may override this size to 1 MB not 64 MB, so use that if so
+    int pageSizeToUse = Integer.parseInt(System.getProperty("spark.buffer.pageSize"));
```
This should just use the value configure in SparkConf? at least that would be the more standard way to plumb through this value to tests.

a-roberts replied:
Yeah that's right, couldn't figure out how to get the SparkConf from that test though, will have another look now

a-roberts replied:
Running the tests now with 
```
int pageSizeToUse = (int) memoryManager.pageSizeBytes(); 
```
Looking at the logic in here I see we either use the default page size or whatever the user specifies

An int is accepted for the allocate method in RowBasedKeyValueBatch, given that a max int in Java is 2,147m I doubt this will be a problem

SparkQA commented:
**[Test build #65319 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/65319/consoleFull)** for PR 15079 at commit [`c793efe`](https://github.com/apache/spark/commit/c793efe6e567ba3fc51ceaf9ff225ca1e5f35f10).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

a-roberts committed:
Use memoryManager.pageSizeBytes

More robust way to get the page size - use either the default value or spark.buffer.pageSize (logic is encapsulated within the method we now use)

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/65319/
Test FAILed.

a-roberts commented:
Above failure (65319) shouldn't happen now as I'm using memoryManager.pageSizeBytes

SparkQA commented:
**[Test build #65323 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/65323/consoleFull)** for PR 15079 at commit [`96a5c66`](https://github.com/apache/spark/commit/96a5c6607e47dd4431f59672f25506181223a93f).

SparkQA commented:
**[Test build #65323 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/65323/consoleFull)** for PR 15079 at commit [`96a5c66`](https://github.com/apache/spark/commit/96a5c6607e47dd4431f59672f25506181223a93f).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/65323/
Test PASSed.

srowen commented:
Merged to master
